src/main/java/core/basesyntax/service/impl/MapTransactionService.java

```
@@ -0,0 +1,18 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.TransactionType;
+import core.basesyntax.service.TransactionHandler;
+import core.basesyntax.service.TransactionService;
+import java.util.Map;
+
+public class MapTransactionService implements TransactionService {
+    private final Map<TransactionType, TransactionHandler> handlers;
+
+    public MapTransactionService(Map<TransactionType, TransactionHandler> handlers) {
+        this.handlers = Map.copyOf(handlers);
+    }
+
+    public TransactionHandler selectHandler(TransactionType transactionType) {
+        return handlers.get(transactionType);
```

`selectHandler` returns `handlers.get(transactionType)` and therefore returns `null` when a mapping is missing. This violates the requirement: "Ensure a handler exists for each operation; if missing, throw an informative exception." Instead of returning null, validate `transactionType` (non-null) and throw an informative RuntimeException when no handler is found (e.g., `throw new RuntimeException("No handler for transaction type: " + transactionType)`). Returning null allows callers (see Database.default `handle`) to attempt to call methods on a null handler and get an NPE instead of a clear, actionable error.